File: swimming-contest-software/Model/src/main/java/domain/enums/SwimmingStyle.java
package domain.enums;

/**
 * Available swimming styles for races.
 */
public enum SwimmingStyle {
    Mixed,
    Freestyle,
    Butterfly,
    Backstroke;

    public static SwimmingStyle styleFromInteger(Integer value) {
        return switch (value) {
            case 1 -> Mixed;
            case 2 -> Freestyle;
            case 3 -> Butterfly;
            case 4 -> Backstroke;
            default -> null;
        };
    }

    public static Integer integerFromStyle(SwimmingStyle swimmingDistance) {
        return switch (swimmingDistance) {
            case Mixed -> 1;
            case Freestyle -> 2;
            case Butterfly -> 3;
            case Backstroke -> 4;
        };
    }
}

File: Model/Domain/Enums/SwimmingStyle.cs
using System;

namespace Model.Domain.Enums;

[Serializable]
public enum SwimmingStyle
{
    Mixed,
    Freestyle,
    Butterfly,
    Backstroke
}

public static class SwimmingStylesMethods
{
    public static SwimmingStyle StyleFromInteger(int value)
    {
        return value switch
        {
            1 => SwimmingStyle.Mixed,
            2 => SwimmingStyle.Freestyle,
            3 => SwimmingStyle.Butterfly,
            4 => SwimmingStyle.Backstroke,
            _ => throw new ArgumentOutOfRangeException(nameof(value), value, null)
        };
    }

    public static int IntegerFromStyle(SwimmingStyle style)
    {
        return style switch
        {
            SwimmingStyle.Mixed => 1,
            SwimmingStyle.Freestyle => 2,
            SwimmingStyle.Butterfly => 3,
            SwimmingStyle.Backstroke => 4,
            _ => throw new ArgumentOutOfRangeException(nameof(style), style, null)
        };
    }
}